src/base/bittorrent/filterparserthread.cpp

```diff
@@ -148,293 +208,111 @@ int FilterParserThread::parseDATFilterFile()
-                start = endOfLine;
-                continue;
-            }
+            const auto& [startIP, endIP] = parseRange(range);
 
-            // Each line should follow this format:
-            // Some organization:1.0.0.0-1.255.255.255
-            // The "Some organization" part might contain a ':' char itself so we find the last occurrence
-            const int partsDelimiter = findAndNullDelimiter(buffer.data(), ':', start, endOfLine, true);
-            if (partsDelimiter == -1)
+            if (startIP.empty() || endIP.empty())
             {
                 ++parseErrorCount;
                 addLog(tr("IP filter line %1 is malformed.").arg(nbLine));
-                start = endOfLine;
                 continue;
             }
 
-            // IP Range should be split by a dash
-            const int delimIP = findAndNullDelimiter(buffer.data(), '-', partsDelimiter + 1, endOfLine);
-            if (delimIP == -1)
-            {
-                ++parseErrorCount;
-                addLog(tr("IP filter line %1 is malformed.").arg(nbLine));
-                start = endOfLine;
-                continue;
-            }
-
-            lt::address startAddr;
-            int newStart = trim(buffer.data(), partsDelimiter + 1, delimIP - 1);
-            if (!parseIPAddress(buffer.data() + newStart, startAddr))
+            std::optional<lt::address> startAddr = parseIPAddress(startIP);
+            if (!startAddr)
             {
                 ++parseErrorCount;
                 addLog(tr("IP filter line %1 is malformed. Start IP of the range is malformed.").arg(nbLine));
-                start = endOfLine;
                 continue;
             }
 
-            lt::address endAddr;
-            newStart = trim(buffer.data(), delimIP + 1, endOfLine);
-            if (!parseIPAddress(buffer.data() + newStart, endAddr))
+            std::optional<lt::address> endAddr = parseIPAddress(endIP);
+            if (!endAddr)
             {
                 ++parseErrorCount;
                 addLog(tr("IP filter line %1 is malformed. End IP of the range is malformed.").arg(nbLine));
-                start = endOfLine;
                 continue;
             }
 
-            if ((startAddr.is_v4() != endAddr.is_v4())
-                || (startAddr.is_v6() != endAddr.is_v6()))
-                {
+            if ((startAddr->is_v4() != endAddr->is_v4())
+                || (startAddr->is_v6() != endAddr->is_v6()))
+            {
                 ++parseErrorCount;
                 addLog(tr("IP filter line %1 is malformed. One IP is IPv4 and the other is IPv6!").arg(nbLine));
-                start = endOfLine;
                 continue;
             }
 
-            start = endOfLine;
-
+            // Now Add to the filter
             try
             {
-                m_filter.add_rule(startAddr, endAddr, lt::ip_filter::blocked);
+                m_filter.add_rule(std::move(*startAddr), std::move(*endAddr), lt::ip_filter::blocked);
```

Comment:
The `add_rule()` takes a const reference, so `std::move` isn't justified.
Personally, I don't like to use `std::move` without a good reason, it is hard to maintain.

Reply:
Sorry, I was hasty at that point in time. I didn't investigate the add_rule() signature.